packages/teams-js/src/public/visualMedia.ts

```diff
@@ -0,0 +1,307 @@
+import { sendAndHandleSdkError } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { errorNotSupportedOnPlatform, FrameContexts } from './constants';
+import { DevicePermission } from './interfaces';
+import { runtime } from './runtime';
+
+/**
+ * Interact with image and video. It lets the app developer ask the user to get images or video from their camera / camera roll / file system.
+ * @beta
+ */
+export namespace visualMedia {
+  /**
+   * @hidden
+   * Hide from docs
+   * --------
+   * All properties common to Image and Video Props
+   * @beta
+   */
+  interface VisualMediaProps {
+    /**
+     * @hidden
+     * Lets the developer specify the visualMedia source
+     */
+    sourceProps: GalleryProps | CameraProps;
+    /**
+     * max limit of media allowed to be selected in one go, current max limit is 10.
+     */
+    maxVisualMediaCount: number;
+  }
+  /**
+   *  All properties in ImageProps are optional and have default values in the platform
+   *  Additional properties for image in mobile devices
+   * @beta
+   */
+  export interface ImageProperties extends VisualMediaProps {
+    /**
+     * Optional; indicate if inking on the selected Image is allowed or not
+     * Default value is true
+     */
+    shouldAllowInkingOnImages?: boolean;
```

Comment:
Thanks for the great point, as web/desktop is not in the P0 in PM's spec. Since it is the feature that support by mobile platform, it is low lift to add those boolean values. When we start on the web/desktop version, and if web/desktop is not supporting those features, I will make sure call that out.

Reply:
> Just wanted to make sure: the way this is set up now there will be no way for web/desktop to choose _not_ to support inking etc. becase it's baked into the new functions you're adding that any developer can ask for them. That might be okay, but if it's not (because some platforms won't support that functionality) then we would have to either restructure some of this api so allow platforms to choose not to support those features, _or_ we'd have to document that features like inking/stickers etc. are best effort and if a developer asks for them on a platform that doesn't support them they will silently not get that functionality.
> 
> Is that okay?

Definitely, and I understand the concern here. I changed the values like inking, stickers, etc., optional so that platforms like desktop and web that don't support them can simply not implement them. Additionally, I've documented which properties are optional and describe the expected behavior if a property is not set or if a platform doesn't support it. For example: If set to true but the platform does not support inking, this feature will be silently ignored. 
How do you feel about it？

Reply:
Synced offline, and remove the ink/textStickers/filter. Please unresolve this comment if needed. Thanks a lot for sharing your thought!